### ISSUE TITLE
Fix request: FieldOfHashCodeMissingInEquals false positive cached hash code

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#503]
```java
@Data
public abstract class Foo<T>  {

    /** Cache the hash code */
    private transient int hash; // Default to 0. <- false positive

    private final String serviceId;
    private final String sessionId;
    private final TenantId tenantId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Foo that = (Foo) o;
        return sessionId.equals(that.sessionId) &&
                tenantId.equals(that.tenantId);
    }

    @Override
    public int hashCode() {
        if (hash == 0) {
            hash = Objects.hash(sessionId, tenantId);
        }
        return hash;
    }
}
```

**Comment by jborgers** [comment]
@stokpop I cannot reproduce this. Are you sure it is a false positive of that rule, MissingFieldInHashCode? That is, hash is not missing in hashCode.

**Comment by jborgers** [comment]
@stokpop Do you mean: FieldOfHashCodeMissingInEquals?

**Comment by stokpop** [comment]
yes you are right: `FieldOfHashCodeMissingInEquals`, updated title
